package/gluon-config-mode-domain-select/luasrc/lib/gluon/config-mode/wizard/0200-domain-select.lua

```diff
@@ -22,10 +21,7 @@ return function(form, uci)
 
 	local function get_domain_list()
 		local list = {}
-		for _, domain_path in ipairs(util.glob('/lib/gluon/domains/*.json')) do
-			local domain_code = domain_path:match('([^/]+)%.json$')
-			local domain = assert(json.load(domain_path))
-
+		for domain_code, domain in pairs(util.get_domains()) do
```

Comment:
This looks wrong, was this even tested?

- `get_domains()` returns an integer-indexed table of entries of the form `{ domain_code = ..., domain = ... }`
- Using `pairs()` you are iterating over the table key-value pairs (in unspecified order), so in this loop `domain_code` will be the integer index of the domain, and `domain` will be the `{ domain_code = ..., domain = ... }` entry

For `gluon-config-mode-domain-select` the optimal solution would be change `get_domains()` to return a table mapping the domain code to the loaded JSON - in that case iterating over it using `pairs()` would do the correct thing (and the unspecified iteration order is fine, as we sort the list afterwards).

The hoodselector would require more changes if `get_domains()` were modified this way (and it can't depend on the alphabetical order of the entries returned by `glob()`).